### ISSUE TITLE
Vector Brush Range undo issue with creating new frames

### ISSUE THREAD

**Issue by manongjohn** [opentoonz/opentoonz#3070]
### Issue Summary
I've encountered some issues when undo/redoing a Vector Brush Range action when you either start with an empty cell  then draw on an exposed cell, vice versa or if you start and end on empty cells.

Basically when drawing start and end frames where both frames already exist, undo/redo works fine.  When drawing starting and end frames but one or both of them starts of empty, the newly created frame when you start drawing either is not undoable  or it deletes all the frames the interpolation is drawn on.

### Steps to Reproduce

OT 1.3 Example - Starting frame is empty:
1. Create a new Vector level with 5 frames with different drawings
2. Expose Level Frames 1, 3-5 in Xsheet Frames 1, 3-5 respectively. Leave Frame 2 empty
3. Change Vector Brush Range to Linear
4. Draw 1st shape in Xsheet Frame 2 (empty) -> Creates and exposes Level Frame 2a
5. Draw 2nd shape in Xsheet Frame 5 -> Interpolation drawn between 2, 2a, 3-5
6. Undo -> Interpolation is undone. 2a remains
7. Undo -> 2a remains exposed. Whatever action was performed before the creation of 2a is undone
8. Keep undoing -> 2a remains, all other actions are undone.
- In 1.4rc  this is especially worse because instead of 2a being created, the existing frame are renumbered and a new Frame 2 is created.  The renumber never gets undone and further undo/redo action are now being applied to the wrong frames.

Example - Ending frame is empty
1. Create a new Vector level with 5 frames with different drawings
2. Expose Level Frames 1-5 in Xsheet
3. Change Vector Brush Range to Linear
4. Draw 1st shape in Xsheet Frame 2
5. Draw 2nd shape in Xsheet Frame 6 (empty) -> Frame 6 created. Interpolation drawn between 2-6
6. Undo -> Frames 2-6 are permanently deleted.  (Dev note: This is likely because the m_isFrameCreated = true is being carried through the drawing of the interpolation on existing frame when it should only apply to the newly created frame and false everywhere else)

As Guided Drawing uses the Range logic, it also suffers from these issues.

**Comment by manongjohn** [comment]
Just realized the second half of the issue I reported here is the same as #1778. As this issue covers 2 Vector Range and frame creation issues with different results but likely similar causes, I recommend closing that one and keeping this.

**Comment by RodneyBaker** [comment]
Current testing:  I get different results for step 6:  Frames 2 - 6 are not deleted.
@manongjohn Can you confirm this problem still occurs for you?
Thanks

**Comment by manongjohn** [comment]
Yes.  Although the undo behavior is different, it is still an issue.    Now, when I undo, the newly created Frame 2 is deleted, but the following frames are not renumbered back. Additionally, the newly created Frame 7 remains, leaving Frame 1, 3-7.  It should have deleted the newly created Frame 2 and Frame 7 and renumbered the frames inbetween to restore it back to Frames 1-5.

**Comment by RodneyBaker** [comment]
Thanks for the revisit @manongjohn 
I'll revisit your steps as I'm very obviously missing something in the process.
